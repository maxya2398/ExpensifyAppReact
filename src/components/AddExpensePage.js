import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
<<<<<<< HEAD
  onSubmit = (expense) =>{
=======
  onSubmit = (expense) => {
>>>>>>> de3a69f3ec1d8c57c8a719084edbb40bf9d15307
    this.props.startAddExpense(expense);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
