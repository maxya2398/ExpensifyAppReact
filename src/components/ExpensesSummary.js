import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import {Link} from 'react-router-dom';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? 'gasto' : 'gastos' ;
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
  
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Viendo <span>{expenseCount}</span> {expenseWord} dando un total <span>{formattedExpensesTotal}</span></h1>
        <div className="page-header__actions">
          <Link className="button" to="/create">Agregar Gasto</Link>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
