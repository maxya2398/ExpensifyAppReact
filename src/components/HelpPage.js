import React from 'react';

const HelpPage = () => (
  <div>
    Esto es de ayuda
  </div>
);

export default HelpPage;
