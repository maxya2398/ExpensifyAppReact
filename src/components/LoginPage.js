import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

export const LoginPage = ({startLogin}) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">App de Gastos</h1>
      <p>Registra todos tus gastos</p>
      <button className="button" onClick={startLogin}>Iniciar Sesi√≥n con Google</button>
    </div>
  </div>
);

const mapDispatchToProps = ( dispatch )=>({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);

// export const LoginPage = ({startLogin}) => (
//   <div className="box-layout">
//     <div className="box-layout__box">
//       <h1 className="box-layout__title">App de Gastos</h1>
//       <p>Registra todos tus gastos</p>
//       <button onClick={startLogin}>Login</button>
//     </div>
//   </div>
// );