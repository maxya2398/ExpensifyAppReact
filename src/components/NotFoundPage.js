import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  <div>
    404 - <Link to="/">Ir a Inicio</Link>
  </div>
);

export default NotFoundPage;
