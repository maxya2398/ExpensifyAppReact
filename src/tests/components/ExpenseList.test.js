import React from 'react';
import {shallow} from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';


test('Render Expense List con expenses', ()=>{
    const wrapper = shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});

test('Vacio Expense List', ()=>{
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
})