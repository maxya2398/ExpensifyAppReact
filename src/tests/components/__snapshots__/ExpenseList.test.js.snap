// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Expense List con expenses 1`] = `
<div>
  <ExpenseListItem
    amount={1.95}
    createdAt={0}
    description="Goma"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={300}
    createdAt={-345600000}
    description="Lapiz"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={18.23}
    createdAt={345600000}
    description="Calculadora"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`Vacio Expense List 1`] = `
<div>
  <p>
    No hay gastos
  </p>
</div>
`;
